apps/src/lib/levelbuilder/AllVocabulariesEditor.jsx

```diff
@@ -0,0 +1,215 @@
+    addVocabulary: PropTypes.func.isRequired,
+    updateVocabulary: PropTypes.func.isRequired,
+    removeVocabulary: PropTypes.func.isRequired,
+
+    courseVersionId: PropTypes.number.isRequired,
+    courseName: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      vocabularyForEdit: null,
+      vocabularyForDeletion: null,
+      addVocabularyDialogOpen: false
+    };
+  }
+
+  getColumns() {
+    return [
+      {
+        property: 'actions',
+        header: {
+          label: 'Actions'
+        },
+        cell: {
+          formatters: [this.actionsCellFormatter],
+          props: {
+            style: {
+              ...lessonEditorTableStyles.actionsCell
+            }
+          }
+        }
+      },
+      {
+        property: 'word',
+        header: {
+          label: 'Word'
+        },
+        cell: {
+          props: {
+            style: {
+              ...lessonEditorTableStyles.cell
+            }
+          }
+        }
+      },
+      {
+        property: 'definition',
+        header: {
+          label: 'Definition'
+        },
+        cell: {
+          props: {
+            style: {
+              ...lessonEditorTableStyles.cell
+            }
+          }
+        }
+      }
+    ];
+  }
+
+  handleEdit = vocabularyForEdit => {
+    this.setState({vocabularyForEdit, addVocabularyDialogOpen: true});
+  };
+
+  actionsCellFormatter = (actions, {rowData}) => {
+    return (
+      <div style={styles.actionsColumn}>
+        <div onMouseDown={() => this.handleEdit(rowData)} style={styles.edit}>
+          <i className="fa fa-edit" />
+        </div>
+        <div
+          onMouseDown={() => this.setState({vocabularyForDeletion: rowData})}
+          style={styles.remove}
+          className={'unit-test-destroy-vocabulary'}
+        >
+          <i className="fa fa-trash" />
+        </div>
+      </div>
+    );
+  };
+
+  afterVocabularySave = vocabulary => {
+    if (this.state.vocabularyForEdit) {
+      this.props.updateVocabulary(vocabulary);
+    } else {
+      this.props.addVocabulary(vocabulary);
+    }
+  };
+
+  handleDeleteVocabularyDialogClose = () => {
+    this.setState({
+      vocabularyForDeletion: null
+    });
+  };
+
+  handleDeleteVocabularyConfirm = () => {
+    this.props.removeVocabulary(this.state.vocabularyForDeletion.key);
+    this.handleDeleteVocabularyDialogClose();
+  };
+
+  handleAddVocabularyClick = e => {
+    e.preventDefault();
+    this.setState({addVocabularyDialogOpen: true});
+  };
+
+  render() {
+    const columns = this.getColumns();
+    return (
+      <div>
+        <div style={styles.header}>
+          <h1>{`Vocabulary for ${this.props.courseName}`}</h1>
+          <a
+            onClick={this.handleAddVocabularyClick}
+            style={styles.addButton}
+            type="button"
+          >
+            <i className="fa fa-plus" style={{marginRight: 7}} /> Create New
```

Comment:
nit: Should "Create New" be on the next line with "Vocabulary"?

Reply:
hmm prettier is breaking it up...I'll just move the whole thing to the next line